### PR TITLE
bpo-32297: Few misspellings found in Python source code comments.

### PR DESCRIPTION
**Pull request by mehanig**
I've extracted all the comments from `*.py` files and re-read them manually with the help of spelling and grammar tools. 
Most of the changes are a misspelling or something like incorrect editing. I wasn't expected to touch intentionally any other files except `*.py`, but while searching through codebase found few mistakes also.

And, few files are antique (10+ years), so I feel very nervous touching that part of the Python history.

https://bugs.python.org/issue32297

### CODE REVIEW

File: Lib/test/test_importlib/source/test_file_loader.py
Comment (mehanig) [inline Lib/test/test_importlib/source/test_file_loader.py:270]: I decided to change function name for the sake of correct spelling. It shouldn't affect any other code and looks like tests aren't failing.

**Comment by Mariatta** [comment]
Hmm, the fixes here are for internal comment, not docstrings or documentation. I'm thinking these should be left as is, unless we are actually fixing those parts of the code.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
Mike, thank you for adding Python to the list of projects  you contribute to.  I presume you carefully checked the changes yourself.  I double-checked and all look correct.  Although not public, they make the code nicer to read. 

I am the maintainer of idlelib, which is a special case, in that I even backport changes like this.  So I will make the 3 changes in a separate patch and then revert them on this PR.  IE, I will do the change I am requesting.  Edit: Done #4807, #4809 

I believe Mariatta's concern is about code churn and blame history.  I will let everyone else decide what to do with non-idlelib changes.

Though each change is trivial, there are enough that I would have preferred that you started with a tracker issue.  But I don't care now.

Test name changes should not affect anything else.  If the test were my test (in idle_test), I would accept the name change, but it is not, so I leave it to the import people.

**Review by terryjreedy — APPROVED** [review]

**Comment by mehanig** [comment]
Thank you, Terry. Yes, I agree that it's might be good to open an Issue and discuss whether or not we should merge this request (partially or entirely). 
I didn't know about idlelib special cases, thank you a lot for helping with it and backporting changes.

Mariatta, you are 100% correct, most of the changes are internal code comments, except one docstring (https://github.com/python/cpython/pull/4803/files#diff-be61682fc3333d3bd8fc466a4e4dbdcfR1261). And I completely agree with you that it might be useful, in some cases, to not touch git blame history. 

I think, at this moment, it's up for maintainers to decline or approve this changes, and I am open to exclude any of them from this request.

**Comment by brettcannon** [comment]
Can you open an issue @mehanig on bugs.python.org and edit the title to reference it so we can decide if we want to accept your PR?

**Comment by mehanig** [comment]
Yes, done. Sorry for my late reply.

**Comment by vstinner** [comment]
@Mariatta: "Hmm, the fixes here are for internal comments, not docstrings or documentation. I'm thinking these should be left as is, unless we are actually fixing those parts of the code."

@terryjreedy: "I believe Mariatta's concern is about code churn and blame history. I will let everyone else decide what to do with non-idlelib changes."

IMHO it's ok to fix typos :-) But I didn't review the change itself. I asked @csabella and @CuriousLearner to take a look.

**Comment by Mariatta** [comment]
Yeah well, it makes me uneasy when a PR touched 41 different files.
I suspect there will be conflict when backporting...
If other core devs think it's worth fixing, I really won't object.
I did not explicitly approve/reject the PR. Simply raising a concern :)

**Comment by vstinner** [comment]
>  I suspect there will be conflict when backporting...

Ah right, we have to backport this PR to Python 3.6 if we choose to merge it, to reduce conflicts when backporting fixes into Python 3.6.

I propose to not backport it to Python 2.7, since Python 2.7 is dying and minor fixes are no more backported to 2.7.

File: Lib/test/test_logging.py
Comment (CuriousLearner) [inline Lib/test/test_logging.py]: If you're fixing this sentence. I would suggest to also leave a space between `#` and the first word for the sentence.

File: Lib/test/test_math.py
Comment (CuriousLearner) [inline Lib/test/test_math.py]: For the comments on the same line as code, please leave 2 spaces before starting the comment.

If we choose to change the git blame for this line, then it makes sense to fix as much as we can :)

**Review by CuriousLearner — COMMENTED** [review]
I think for the spell fixes in comment and test cases are okay, since it won't affect other parts.

On the other hand, I agree that it may cause some issue in back-porting it to `3.6`.

Apart from that, I have just minor comments on things. If this change is merged in, it would change the git blame for those lines, so it makes sense to do these changes too.

File: Lib/test/test_importlib/source/test_file_loader.py
Comment (CuriousLearner) [reply Lib/test/test_importlib/source/test_file_loader.py:270]: I think this is okay since it won't affect anything else, and test can run as it is.

**Comment by csabella** [comment]
I agree with @CuriousLearner 's suggestions and the other changes LGTM.

If it helps, 26 of the 41 file changes are for test files, so I don't know if the blame history is as important on those.  Some of the changes to code files are ones that have been modified a few times this year, like asyncio, http, os, xml, and faulthandler.c.  Maybe the maintainers of those should make individual decisions for blame, while the rest of the module changes might not be happen often enough to worry about the history.

**Comment by asvetlov** [comment]
#4839 for asyncio typos

**Comment by mehanig** [comment]
@asvetlov Thank you. Is it correct, that I need now to exclude from this request the changes you picked?

File: Lib/test/test_math.py
Comment (mehanig) [reply Lib/test/test_math.py]: Good point, done in https://github.com/python/cpython/pull/4803/commits/e2321c944031c8eeb87b7132e70e23c02900f20e

**Comment by asvetlov** [comment]
Just merge master branch from upstream

**Review by CuriousLearner — COMMENTED** [review]
Changes you did till 
e2321c9, LGTM.

I don't think changes done in 53592fc are necessary.

Since we were fixing typos in previous comments, it made sense to also correctly format the comments.

The module owner can take a decision if the diff in 53592fc is really needed :)

Thanks for your work :)

File: Lib/test/test_logging.py
Comment (mehanig) [reply Lib/test/test_logging.py]: Yes, I agree. I even fixed all the other places in this file. Otherwise, it might look very inconsistent. I'm talking about this last commit https://github.com/python/cpython/commit/53592fc0b8ba08f984de36cd5bd266e13ec0ac27, but it might be overhead and I'm open to revert it.

**Comment by mehanig** [comment]
@CuriousLearner Good! Yes, I'm also not sure about the last one :)

**Comment by brettcannon** [comment]
FYI I approve of the importlib test changes.

File: Lib/idlelib/autocomplete_w.py
Comment (JelleZijlstra) [inline Lib/idlelib/autocomplete_w.py]: Here it looks like the PR is introducing a typo. Did something go wrong with the separate idlelib change?

Comment (mehanig) [reply Lib/idlelib/autocomplete_w.py]: Thank you, my bad, incorrect rebase. Since current master already merged that changes separately, I now excluded idlelib revert-patches from this request.

**Review by rhettinger — APPROVED** [review]

**Review by asvetlov — APPROVED** [review]

**Comment by asvetlov** [comment]
Thanks!

**Comment by vstinner** [comment]
@mehanig: Can you please backport your fix to Python 3.6? The backport will prevent conflicts when other changes will have to be backported.

**Comment by asvetlov** [comment]
Backport to 3.6 has merge conflicts, let's not touch the branch.

**Comment by vstinner** [comment]
> Backport to 3.6 has merge conflicts, let's not touch the branch.

Please review previous comments starting near:
https://github.com/python/cpython/pull/4803#issuecomment-351240925

This PR must be backported to Python 3.6. @mehanig (or someone else) can easily fix conflicts.

**Comment by asvetlov** [comment]
Ok, I did it: #4862

**Comment by mehanig** [comment]
@asvetlov Thank you so much for your help!

**Pull request merged**
